### PR TITLE
Scala 3: given imports, pattern-bound givens

### PR DESCRIPTION
**Pull request by MichalBednarz**
#2216

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala [hunk 1]
```diff
@@ -129,8 +129,11 @@ class Router(formatOps: FormatOps) {
           Split(NoSplit, 0)
         )
       // Import
-      case FormatToken(_: T.Dot, _: T.LeftBrace | _: T.Underscore, _)
-          if existsParentOfType[Import](rightOwner) =>
+      case FormatToken(
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala:132]: should this just be `case FormatToken(_: T.Dot, _, _)`?

**Review by kitbellew — CHANGES_REQUESTED** [review]

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala (hunk 1 above)
Comment (MichalBednarz) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala]: good point, it works as well!

**Review by tgodzik — APPROVED** [review]
LGTM!

File: scalafmt-tests/src/test/resources/scala3/ImportingGivens.stat [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+<<< merged import clauses
+object B {
+  import A.{given, _}
+}
+>>>
+object B {
+  import A.{given, _}
+}
+<<< import given of type
+maxColumn = 20
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/ImportingGivens.stat:20]: can you add the same but with 10?

File: scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat:1]: this one is not related to a code change, right?

File: scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+for given Context <- applicationContexts yield ()
+<<< for comprehension with parens
+for (given Context <- applicationContexts)
+yield ()
+>>>
+for (given Context <- applicationContexts)
+  yield ()
+<<< match case
+context match {
+  case (ctx @ given Context) => ctx
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat:14]: can you add two more cases, with shorter max column, to force break before and after given? indent is of interest.

**Review by kitbellew — CHANGES_REQUESTED** [review]
my apologies, previous review was incomplete, i forgot something

File: scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat (hunk 4 above)
Comment (MichalBednarz) [reply scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat:13]: do we want the given to break in such a situation? in the case of `case (ctx @ Context) => ctx` the line breaks only after the arrow.

File: scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat (hunk 3 above)
Comment (MichalBednarz) [reply scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat]: right, for test purposes only

File: scalafmt-tests/src/test/resources/scala3/ImportingGivens.stat (hunk 2 above)
Comment (MichalBednarz) [reply scalafmt-tests/src/test/resources/scala3/ImportingGivens.stat:19]: looks same, corresponds with the situation without given

File: scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat (hunk 4 above)
Comment (kitbellew) [reply scalafmt-tests/src/test/resources/scala3/GivenPatterns.stat:13]: I thought it would break in the middle of the `x @ y` expression...

File: scalafmt-tests/src/test/resources/scala3/ImportingGivens.stat [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  given T2,
+  given T3
+}
+<<< import given of type doesn't break
+maxColumn = 10
+===
+import A.{given T1, given T2, given T3}
+>>>
+import A.{
+  given T1,
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/ImportingGivens.stat:34]: Looks like it doesn't break in the middle... Well, that's fine.

**Review by kitbellew — APPROVED** [review]
